build_tools/android/run_benchmarks.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Runs all matched benchmark suites on an Android device.
+
+This script probes the Android phone via `adb` and uses the device information
+to filter and run suitable benchmarks on it.
+
+It expects that `adb` is installed, and there is an `iree-benchmark-module`
+tool cross-compiled towards Android. It also expects the benchmark artifacts
+are generated by building the `iree-benchmark-suites` target.
+
+Example usages:
+  python3 run_benchmarks.py \
+    --benchmark_tool=/path/to/android/target/iree-benchmark_module \
+    /path/to/host/build/dir
+"""
+
+import argparse
+import json
+import os
+import re
+import subprocess
+
+from common.benchmark_description import AndroidDeviceInfo, BenchmarkInfo, BenchmarkResults
+
+# Relative path against build directory.
+BENCHMARK_SUITE_REL_PATH = "benchmark_suites"
+# Relative path against root benchmark suite directory.
+PYTON_MODEL_REL_PATH = "tf_models"
+
+# The flagfile's filename for compiled Python models.
+MODEL_FLAGFILE_NAME = "flagfile"
+# The artifact's filename for compiled Python models.
+MODEL_VMFB_NAME = "compiled.vmfb"
+
+# Root directory to perform benchmarks in on the Android device.
+ANDROID_TMP_DIR = "/data/local/tmp/iree-benchmarks"
+
+BENCHMARK_REPETITIONS = 10
+
+# A map from Android CPU ABI to IREE's benchmark target architecture.
+CPU_ABI_TO_TARGET_ARCH_MAP = {
+    "arm64-v8a": "cpu-arm64-v8a",
+}
+
+# A map from Android GPU name to IREE's benchmark target architecture.
+GPU_NAME_TO_TARGET_ARCH_MAP = {
+    "adreno-640": "gpu-adreno",
+    "adreno-650": "gpu-adreno",
+    "adreno-660": "gpu-adreno",
+    "mali-g77": "gpu-mali-valhall",
+    "mali-g78": "gpu-mali-valhall",
+}
+
+
+def execute(args,
```

Comment:
Import from common? Although then my comment about "all of these are the same" isn't actually true. We have many "abstractions" over `subprocess.run` and I'm not really sure how much they add, since they're basically just taking different opinions about the defaults and at least `subprocess.run` is the official one. Then again, logging the command seems useful and "universal_newlines" is a totally bonkers name for that kwarg.